File: src/main/java/com/windea/demo/csntportal/service/ReformColumnService.java
package com.windea.demo.csntportal.service;

import com.windea.demo.csntportal.domain.entity.ReformColumn;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

/**
 * 教学改革专栏的服务接口。
 */
public interface ReformColumnService {
	ReformColumn create(ReformColumn column);

	void delete(Integer id);

	ReformColumn update(ReformColumn column);

	ReformColumn get(Integer id);

	Page<ReformColumn> list(Pageable pageable);

	Page<ReformColumn> searchByTitle(String title, Pageable pageable);
}

File: src/main/mobile/src/app/service/api/reform-column.service.ts
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {ReformColumn} from "../../domain/entity/ReformColumn";
import {Observable} from "rxjs";
import {apiUrl} from "../../../environments/environment.prod";
import {Page} from "../../domain/interface/Page";

/**
 * 教学改革专栏的服务类。
 */
@Injectable({providedIn: "root"})
export class ReformColumnService {
  constructor(private http: HttpClient) {
  }


  create(column: ReformColumn): Observable<ReformColumn> {
    const url = `${apiUrl}/reform-column/create`;
    return this.http.post<ReformColumn>(url, column);
  }

  delete(id: number) {
    const url = `${apiUrl}/reform-column/${id}`;
    return this.http.delete(url);
  }

  update(column: ReformColumn): Observable<ReformColumn> {
    const url = `${apiUrl}/reform-column/${column.id}`;
    return this.http.put<ReformColumn>(url, column);
  }

  get(id: number): Observable<ReformColumn> {
    const url = `${apiUrl}/reform-column/${id}`;
    return this.http.get<ReformColumn>(url);
  }

  list(page: number, size: number): Observable<Page<ReformColumn>> {
    const url = `${apiUrl}/reform-column/list`;
    const params = {page: page + "", size: size + ""};
    return this.http.get<Page<ReformColumn>>(url, {params: params});
  }

  searchByTitle(title: string, page: number, size: number): Observable<Page<ReformColumn>> {
    const url = `${apiUrl}/reform-column/search`;
    const params = {title: title, page: page + "", size: size + ""};
    return this.http.get<Page<ReformColumn>>(url, {params: params});
  }
}
